radicle-node/src/service.rs

```diff
@@ -891,14 +915,46 @@ where
                         match message.is_fresh(&self.storage) {
                             Ok(is_fresh) => {
                                 if is_fresh {
-                                    // TODO: Only fetch if the refs announced are for peers we're tracking.
-                                    self.fetch(message.rid, announcer);
+                                    match repo_policy.scope {
+                                        tracking::Scope::Trusted => {
+                                            // Check every node and only fetch if we're tracking the node.
+                                            let doc = self.get(message.rid)?;
+
+                                            let trusted: Vec<_> = message.refs.iter().filter_map(|(pub_key, _refs)| {
+                                                let is_delegate = match &doc {
+                                                    Some(d) => d.is_delegate(pub_key),
+                                                    None => false,
+                                                };
+                                                let is_node_tracked = self
+                                                    .tracking.is_node_tracked(pub_key)
+                                                    .expect("handle_announcement: error accessing node tracking configuration");
+                                                if is_node_tracked || is_delegate {
+                                                    Some(*pub_key)
+                                                } else {
+                                                    debug!("handle_announcement: skip fetch as we're not tracking {pub_key}");
+                                                    None
+                                                }
+                                            }).collect();
+                                            if let Some(non_empty) = NonEmpty::from_vec(trusted) {
+                                                self.fetch(
+                                                    message.rid,
+                                                    Namespaces::Many(non_empty),
+                                                    announcer,
+                                                );
+                                            } else {
+                                                debug!("handle_announcement: no trusted nodes to fetch");
```

Comment:
Logging is only useful if you're going to include some context :)

```suggestion
                                                debug!("handle_announcement: no trusted nodes to fetch {rid} from;");
```

Reply:
Here we also should also probably respond to any `fetch_reqs`

Reply:
> Here we also should also probably respond to any `fetch_reqs`

I missed that. I'm wondering why do we need to respond to `fetch_reqs` here in handling `refs_announcement`?